dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java

```diff
@@ -3,15 +3,150 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
+import java.io.*;
 
 public class ResultsParser {
 
+
+    public static final String PARALLELIZATION = "New order time:"; //name of line to get values from in Parallelization technique
+    public static final String APFD_VALUE = "APFD value:"; //name of line to get values from in selection and prioritization
+    
+    /*
+     * A method to search a file for a keyword and return the value that follows that keyword
+     * @return the data value without any leading or trailing whitespaces
+     */
+    public static String parseFile(File file, String keyword) throws FileNotFoundException{
+        
+        Scanner scanner = new Scanner(file); 
+        while(scanner.hasNextLine()){
+            String currLine = scanner.nextLine();
+            //lineNum++;
+            if(currLine.contains(keyword)){
+                String data = currLine.substring(keyword.length(), currLine.length()); //gets numeric value of data
+                scanner.close(); //close Scanner before returning
+                return data.trim(); //trim away any whitespaces leading or after the data value
+                
+            }
+        }
+        scanner.close();
+        return null; //none of the lines contained the keyword
+    }
+
     /**
      * @param args
      */
-    public static void main(String[] args) {
+    public static void main(String[] args) { //args: technique, directory...optional args
         List<String> argsList = new ArrayList<String>(Arrays.asList(args));
-        System.out.println(argsList);
+        //System.out.println(argsList);
+
+        String technique = args[0];
```

Comment:
Instead of assuming that the first argument is always technique and the second one is always the directory, look for where in the argument list is a argument like "-techniqueName" and then parse the string after that argument.
For more examples of this see the Wrapper class.
https://github.com/winglam/dependent-tests-impact/blob/master/dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/Wrapper.java